plugins/redfish/fu-plugin-redfish.c

```diff
@@ -122,19 +140,47 @@ fu_redfish_plugin_discover_smbios_table (FuPlugin *plugin, GError **error)
 	hostname = g_strdup (fu_redfish_smbios_get_ip_addr (redfish_smbios));
 	if (hostname == NULL)
 		hostname = g_strdup (fu_redfish_smbios_get_hostname (redfish_smbios));
-	if (hostname == NULL) {
+	if (device == NULL) {
 		const gchar *mac_addr = fu_redfish_smbios_get_mac_addr (redfish_smbios);
 		if (mac_addr != NULL) {
-			hostname = fu_redfish_network_ip_for_mac_addr (mac_addr, error);
-			if (hostname == NULL)
+			device = fu_redfish_network_device_for_mac_addr (mac_addr, error);
+			if (device == NULL)
 				return FALSE;
 		}
 	}
-	if (hostname == NULL) {
+	if (device == NULL) {
 		guint16 vid = fu_redfish_smbios_get_vid (redfish_smbios);
 		guint16 pid = fu_redfish_smbios_get_pid (redfish_smbios);
 		if (vid != 0x0 && pid != 0x0) {
-			hostname = fu_redfish_network_ip_for_vid_pid (vid, pid, error);
+			device = fu_redfish_network_device_for_vid_pid (vid, pid, error);
+			if (device == NULL)
+				return FALSE;
+		}
+	}
+	if (device == NULL) {
+		g_autofree gchar *device_path = NULL;
+		device_path = fu_redfish_plugin_discover_device_path (plugin);
+		if (device_path != NULL) {
+			device = fu_redfish_network_device_for_path (device_path, error);
+			if (device == NULL)
+				return FALSE;
+		}
+	}
+
+	/* autoconnect device if required */
+	if (device != NULL) {
```

Comment:
shouldn't the device == NULL be a failure here that you return false?

Reply:
I don't think so; I think if you just specify an IP address in the SMBIOS blob and not a MAC or VID/PID. I've got a fixup patch I can push on Thursday that removes the RedfishNetworkDevicePath as Lenovo pointed out the VID/PID parsing in the plugin was wonky. Until then I'm in daddy mode for a couple of days.